File: GearUp.Core/Entities/Product.cs
using GearUp.Core.Enum;
using System;
using System.Collections.Generic;
using System.Text;

namespace GearUp.Core.Entities
{
    public class Product : BaseEntity
    {
        public required string Name { get; set; }
        public required string Description { get; set; }
        public decimal Price { get; set; }
        public required string PictureUrl { get; set; }
        public required string Type { get; set; }
        public required string Brand { get; set; }
        public int QuantityInStock { get; set; }

        //vendor attributes
        public ProductStatus Status { get; set; } = ProductStatus.Approved;
        public string? VendorId { get; set; } // optional
        public AppUser? Vendor { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        /// Checks if the product is new (created within the last 7 days)
        public bool IsNew()
        {
            return CreatedAt >= DateTime.UtcNow.AddDays(-7);
        }
    }
}

File: client/src/app/core/models/product.ts
export interface ProductVendor {
  firstName: string | null;
  lastName: string | null;
  userName: string;
  email: string;
}

export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  pictureUrl: string;
  type: string;
  brand: string;
  quantityInStock: number;
  status: 'Pending' | 'Approved' | 'Rejected';
  vendorId: string | null;
  vendor: ProductVendor | null;
  createdAt: string;
}